tests/test_articlehistory.py

```diff
@@ -29,5 +30,76 @@ def test___init__():
     with pytest.raises(KeyError):
         raise KeyError("Data not found")
 
+
+def test_filter_by_tags():
+    '''tests filter by tags function'''
+    art = ArticleHistory("Techno", None, None, ["mobile edit"])
+    assert art.titles == "Techno"
+    assert art.user is None
+    assert art.keyword is None
+    assert art.tags == ["mobile edit"]
+    assert len(art.revisions) != 0
+
+    art2 = ArticleHistory("", None, None, [""])
+    assert art2.titles == ""
+    assert art2.user is None
+    assert art2.keyword is None
+    assert art2.tags == [""]
+    assert len(art2.revisions) == 0
+
+    art3 = ArticleHistory("Salsa (Mexican Cuisine)", None, None, [""])
+    assert art3.titles == "Salsa (Mexican Cuisine)"
+    assert art3.user is None
+    assert art3.keyword is None
+    assert art3.tags == [""]
+    assert len(art3.revisions) == 0
+
+    art4 = ArticleHistory("Cat", "Greatgiant19", None, ["mw-reverted"])
```

Comment:
See comment on line 36 - test should be consistent (eventually this user may have created enough new revisions that the one(s) we're expecting here will fall out of the scope of our default number retrieved)